### PR TITLE
Python 3: xrange -> range

### LINKED ISSUE #9078: Python 3: use native range function instead of six.moves.range for xrange/range differences
Hi,

Another syntax issue to resolve:

### Is your feature request related to a problem? Please describe.
Python 3 unifies range iteration by renaming xrange to range. For code that relies on xrange, it will fail in Python 3 unless edited to call range function.

### Describe the solution you'd like
There are two possibilites:

1. Native: convert xrange() to range().
2. Compatbility shim: use six.moves.range for iteration, and in case a list is needed, wrap tthe call inside a list call.

I propose going with the second solution for now.

### Describe alternatives you've considered
The native mode alternative doesn't require six module. However, because some community add-ons may not be updated to support Python 3 right away, it would be beneficial to provide guidance through use of six module.

### Additional context
This is one of the key differences between Python 2 and 3 and must be resolved before full switch to Python 3 takes place.

Thanks.

### PR DESCRIPTION
**Pull request by josephsl**
### Link to issue number:
Fixes #9078 

### Summary of the issue:
Python 3 replaces xrange function with range.

### Description of how this pull request fixes the issue:
Every call to xrange function has been replaced with range function, along with adding explanatory comments for every occurrence for ease of review and understanding context.

Specifically:

1. Searching for occurrence of xrange (grep -lr "xrange" source).
2. For each occurrence, note down the context. Two ways to do this: looking at it before changing xrange to range, or edit it, use git diff, and look at context.
3. For some of these, add an explanatory note based on context.

For all this work, WSL version of Git and Grep are used (I'm using standard Grep).

### Testing performed:
Tested with Python 3 version of NVDA (source code only), as this is based on a cherry-pick from a source code branch.

### Known issues with pull request:
Slight performance issues as Python 2's range function returns a list, but once Python 3 is introduced, it won't be a problem. Note that there might be some fragments that will expect a list instead of an iterator, so that must be taken care of if found.

### Change log entry:
None

### CODE REVIEW

**Comment by josephsl** [comment]
Hi,

Note that even though this change is trivial, explanatory comments were added for sake of completeness. These comments should remain for a while after the transition is complete (perhaps to be removed once everyone believes transition was a success).

Thanks.

File: source/IAccessibleHandler.py [hunk 1]
```diff
@@ -991,7 +991,7 @@ def getRecursiveTextFromIAccessibleTextObject(obj,startOffset=0,endOffset=-1):
 	except:
 		return text
 	textList=[]
-	for i in xrange(len(text)):
+	for i in range(len(text)):
```
Comment (leonardder) [inline source/IAccessibleHandler.py]: I think something like this should be considered for this case:

```suggestion
	for i, t in enumerate(text):
```

**Review by LeonarddeR — COMMENTED** [review]
I have found a current use of python 2 range that really should be a list. In NVDAObjects.UIA.edge:
`	levels=range(1,7)`

In textInfos.offsets.OffsetsTextInfo._get_boundingRects, the use of range was intentional and can be preserved.

I don't know of any other intentional range uses. I used `git grep "\Wrange("` to find current uses of range.

**Comment by josephsl** [comment]
Hi, let me write review comments then, along with doing something about intentional use of range. Thanks.

**Review by feerrenrut — APPROVED** [review]
Could you update the description to mention how you found all the instances of xrange. I assume you used `git grep`? Please include the regex in the description. While reviewing this we need to consider if we think all usages were found. We also want to know how to repeat this process after merging from master.

Please also mention in the description that range can be used directly in a for loop, and otherwise should be wrapped in a list.

**Pull request merged**